TeamCode/src/main/java/org/firstinspires/ftc/teamcode/systems/distance/DistanceSystem.java

```diff
@@ -71,224 +72,155 @@ public double getFrontRightDistance() {
-            telemetry.log("MecanumDriveSystem","targetPos motorBR: " + driveSystem.motorBackRight.getTargetPosition());
-
-            telemetry.log("MecanumDriveSystem","currentPos motorFL: " + driveSystem.motorFrontLeft.getCurrentPosition());
-            telemetry.log("MecanumDriveSystem","currentPos motorFR: " + driveSystem.motorFrontRight.getCurrentPosition());
-            telemetry.log("MecanumDriveSystem","currentPos motorBL: " + driveSystem.motorBackLeft.getCurrentPosition());
-            telemetry.log("MecanumDriveSystem","currentPos motorBR: " + driveSystem.motorBackRight.getCurrentPosition());
+            correctToFollowWall(closeBuffer, farBuffer, power, StopCondition.DISTANCE);
 
             double direction = 1.0;
             if (distance < 0) {
                 distance = -distance;
                 direction = -1.0;
             }
 
-            double scaledPower = ramp.scaleX(distance);
-            telemetry.log("MecanumDriveSystem","power: " + scaledPower);
+            double scaledPower = shouldRamp ? ramp.scaleX(distance) : power;
+            telemetry.log("DistanceSystem",
+                    "ticks left (ticks): " + driveSystem.getMinDistanceFromTarget());
+            telemetry.log("DistanceSystem","scaled power: " + scaledPower);
+            telemetry.log("DistanceSystem", "distance1: " + getDistance1());
+            telemetry.log("DistanceSystem", "distance2: " + getDistance2());
             driveSystem.setPower(direction * scaledPower);
-            telemetry.log("MecanumDriveSystem","power motorFL: " + driveSystem.motorFrontLeft.getPower());
-            telemetry.log("MecanumDriveSystem","power motorFR: " + driveSystem.motorFrontRight.getPower());
-            telemetry.log("MecanumDriveSystem","power motorBL: " + driveSystem.motorBackLeft.getPower());
-            telemetry.log("MecanumDriveSystem","power motorBR: " + driveSystem.motorBackRight.getPower());
             telemetry.write();
         }
-        driveSystem.motorBackLeft.setPower(0);
-        driveSystem.motorBackRight.setPower(0);
-        driveSystem.motorFrontRight.setPower(0);
-        driveSystem.motorFrontLeft.setPower(0);
+        driveSystem.setPower(0);
+        telemetry.write();
+    }
+
+    public void correctToFollowWall(double closeBuffer, double farBuffer, double correctionPower,
+                                    StopCondition stopCondition) {
+        boolean isOutOfBounds = isOutOfBounds(closeBuffer, farBuffer);
+        if (isOutOfBounds) {
+            driveSystem.setPower(0);
+            telemetry.log("driveTest", "distance buffer triggered");
+        }
+        while (isOutOfBounds(closeBuffer, farBuffer)) {
+            telemetry.log("driveTest", "isOutOfBounds");
+
+            /*boolean reachedStopCondition = false;
+            switch (stopCondition) {
+                case DISTANCE:
+                    reachedStopCondition = (driveSystem.getMinDistanceFromTarget() < 50);
```

Comment:
Remove commented code if it is not going to be used

Reply:
That commented code is going to be used it just has a bug in it. But theoretically it is important

Reply:
If it's unused, stash it or put it in another branch and commit it later. Master shouldn't have any experimental or untested code. (Even if it's commented out.)

Reply:
Good point. I'll take it out.